apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java

```diff
@@ -32,31 +33,62 @@
     if (isFirstLineNum(data)) {
       McsClaim.Builder claimBuilder =
           McsClaim.newBuilder()
-              // Prefixed ICN numbers with '*' to designate synthetic data
-              .setIdrClmHdIcn(icn)
-              .setIdrStatusCodeEnum(McsStatusCode.STATUS_CODE_ACTIVE_A) // Not generated
-              .setIdrBillProvNpi("0000000000") // Not generated
-              .setIdrTotBilledAmt(data.get(Mcs.NCH_CARR_CLM_SBMTD_CHRG_AMT).orElse(""))
+              .setIdrClmHdIcn(
+                  // TODO: Do we need a switch?
+                  ifNull(data.get(Mcs.CARR_CLM_CNTRL_NUM).orElse(null), () -> convertIcn(data)))
               .setIdrClaimMbi(mbiMap.get(beneId).getMbi())
-              .setIdrContrId("00000") // Not generated
-              .setIdrClaimTypeUnrecognized("?") // Not generated
-              .setIdrBillProvNum("0000000000"); // Not generated
+              // Not generated
+              .setIdrBillProvEin("XX-XXXXXXX")
+              .setIdrBillProvNum("0000000000") // TODO: Still valid???
+              .setIdrBillProvSpec("01")
+              .setIdrBillProvType("20")
+              .setIdrClaimReceiptDate("1970-01-01")
+              .setIdrClaimTypeEnum(McsClaimType.CLAIM_TYPE_MEDICAL) // "3"
+              .setIdrContrId("00000") // TODO: Still valid???
+              .setIdrStatusCodeEnum(McsStatusCode.STATUS_CODE_ACTIVE_A)
+              .setIdrStatusDate("1970-01-01");
 
-      claimBuilder.addMcsDetails(
-          McsDetail.newBuilder()
-              .setIdrProcCode(data.get(Mcs.HCPCS_CD).orElse("00000"))
-              .setIdrDtlToDate(data.getFromType(Mcs.CLM_THRU_DT, Parser.Data.Type.DATE).orElse(""))
-              .build());
+      consumeIfNotNull(
+          mbiMap.get(beneId).getFirstName(),
+          value -> claimBuilder.setIdrBeneFirstInit(value.substring(0, 1)));
+      consumeIfNotNull(
+          mbiMap.get(beneId).getLastName(),
+          value -> claimBuilder.setIdrBeneLast16(String.format("%.6s", value)));
+      consumeIfNotNull(
+          mbiMap.get(beneId).getMidName(),
+          value -> claimBuilder.setIdrBeneMidInit(value.substring(0, 1)));
+      // TODO: What is the RIF -> RDA conversion for this value?
+      consumeIfNotNull(
+          mbiMap.get(beneId).getGender(),
+          value -> claimBuilder.setIdrBeneSexEnumValue(Integer.parseInt(value)));
+      data.getFromType(Mcs.CLM_FRM_DT, Parser.Data.Type.DATE)
+          .ifPresent(claimBuilder::setIdrHdrFromDos);
+      data.getFromType(Mcs.CLM_THRU_DT, Parser.Data.Type.DATE)
+          .ifPresent(claimBuilder::setIdrHdrToDos);
+      data.get(Mcs.NCH_CARR_CLM_SBMTD_CHRG_AMT).ifPresent(claimBuilder::setIdrTotBilledAmt);
+      data.get(Mcs.ORG_NPI_NUM).ifPresent(claimBuilder::setIdrBillProvNpi);
 
-      addDiagnosisCodes(claimBuilder, data, icn);
+      McsDetail.Builder detailBuilder = McsDetail.newBuilder();
+
+      data.get(Mcs.LINE_ICD_DGNS_CD).ifPresent(detailBuilder::setIdrDtlPrimaryDiagCode);
```

Comment:
I know we have a check to only process the first line item right now, but in the future if we want to process more line numbers, do you have an idea as to how we could consolidate multiple carrier rows into 1 RDA claim?

Reply:
I've thought about it a little, maybe a cache, maybe a look-ahead, I don't have a solid plan atm, I'd have to get into the implementation a bit to see what seems to work best.